pkg/cloudprovider/provider/scaleway/provider.go

```diff
@@ -0,0 +1,461 @@
+		}
+	}
+
+	ctx := context.TODO()
+	api, err := c.getInstanceAPI()
+	if err != nil {
+		return nil, err
+	}
+
+	imageName, err := getImageNameForOS(pc.OperatingSystem)
+	if err != nil {
+		return nil, cloudprovidererrors.TerminalError{
+			Reason:  common.InvalidConfigurationMachineError,
+			Message: fmt.Sprintf("Failed to parse MachineSpec, invalid operating system specified %q: %v", pc.OperatingSystem, err),
+		}
+	}
+	createServerRequest := &instance.CreateServerRequest{
+		Image:          imageName,
+		Name:           machine.Spec.Name,
+		CommercialType: c.CommercialType,
+		Tags:           append(c.Tags, string(machine.UID)),
+		EnableIPv6:     c.IPv6,
+	}
+
+	serverResp, err := api.CreateServer(createServerRequest, scw.WithContext(ctx))
+	if err != nil {
+		return nil, scalewayErrToTerminalError(err)
+	}
+
+	err = api.SetServerUserData(&instance.SetServerUserDataRequest{
+		Key:      "cloud-init",
+		ServerID: serverResp.Server.ID,
+		Content:  strings.NewReader(userdata),
+	})
+	if err != nil {
+		return nil, scalewayErrToTerminalError(err)
+	}
+
+	klog.V(6).Infof("Scaleway server (id='%s') got fully created", serverResp.Server.ID)
+
+	_, err = api.ServerAction(&instance.ServerActionRequest{
+		Action:   instance.ServerActionPoweron,
+		ServerID: serverResp.Server.ID,
+	})
+	if err != nil {
+		return nil, scalewayErrToTerminalError(err)
+	}
+
+	klog.V(6).Infof("Scaleway server (id='%s') started", serverResp.Server.ID)
+
+	return &scwServer{server: serverResp.Server}, err
+}
+
+func (p *provider) Cleanup(machine *v1alpha1.Machine, _ *cloudprovidertypes.ProviderData) (bool, error) {
+	i, err := p.get(machine)
+	if err != nil {
+		if err == cloudprovidererrors.ErrInstanceNotFound {
+			return true, nil
+		}
+		return false, err
+	}
+
+	c, _, err := p.getConfig(machine.Spec.ProviderSpec)
+	if err != nil {
+		return false, cloudprovidererrors.TerminalError{
+			Reason:  common.InvalidConfigurationMachineError,
+			Message: fmt.Sprintf("Failed to parse MachineSpec, due to %v", err),
+		}
+	}
+	ctx := context.TODO()
+	api, err := c.getInstanceAPI()
+	if err != nil {
+		return false, err
+	}
+
+	_, err = api.ServerAction(&instance.ServerActionRequest{
+		Action:   instance.ServerActionTerminate,
+		ServerID: i.ID(),
+	}, scw.WithContext(ctx))
+	if err != nil {
+		return false, scalewayErrToTerminalError(err)
+	}
+
+	return false, nil
+}
+
+func (p *provider) Get(machine *v1alpha1.Machine, _ *cloudprovidertypes.ProviderData) (cloudInstance.Instance, error) {
+	c, _, err := p.getConfig(machine.Spec.ProviderSpec)
+	if err != nil {
+		return nil, cloudprovidererrors.TerminalError{
+			Reason:  common.InvalidConfigurationMachineError,
+			Message: fmt.Sprintf("Failed to parse MachineSpec, due to %v", err),
+		}
+	}
+
+	api, err := c.getInstanceAPI()
+	if err != nil {
+		return nil, err
+	}
+
+	i, err := p.get(machine)
+	if err != nil {
+		return nil, err
+	}
+
+	switch i.server.State {
```

Comment:
This switch can be simplified to: 
```go
	if i.server.State == instance.ServerStateStopped || i.server.State == instance.ServerStateStoppedInPlace {
		_, err := api.ServerAction(&instance.ServerActionRequest{
			Action:   instance.ServerActionPoweron,
			ServerID: i.server.ID,
		})
		if err != nil {
			return nil, scalewayErrToTerminalError(err)
		}


		return nil, fmt.Errorf("scaleway instance %s is in a stopped state, powering the instance on is in progress", i.Name())
	}
	
	return i, nil
```

As you are not getting anything new after the power up.